Update RBAC principal type documentation (#18859)
DataDog/documentation: content/en/account_management/rbac/_index.md

## Restrict access to individual resources

> Some resources allow you to restrict access to individual resources by roles, teams (beta), or users (beta). We recommend that you use teams to map access to functional groups in your organizations (ex: restrict editing of a dashboard to the application team owning it), roles to map access to personas (ex: restrict editing of a dashboard to administrators), and users only when necessary to encourage knowledge sharing and collaboration.

Reviewer: ```suggestion
Some resources allow you to restrict access to individual resources by roles, teams (beta), or users (beta). Datadog recommends that you use teams to map access to functional groups in your organizations (for example, restrict editing of a dashboard to the application team owning it), roles to map access to personas (for example, restrict editing of a dashboard to administrators), and users only when necessary. This encourages knowledge sharing and collaboration.
```
